### PR TITLE
feat: keep search and hash URL parts when navigating

### LINKED ISSUE #261: Anchors with query parameters or hashes dont work
In the current setup, anchors with query parameters or hashes do not work:
```
<a href="/foo?foo=bar">foo</a>
<a href="/foo#foo=bar">foo</a>
```

Clicking on these anchors, both route to /foo without the query params or hash.

Even if query params are not currently supported as a path to match on, it should at least still be reflected to the URL.

### PR DESCRIPTION
**Pull request by platosha**
Fixes #261

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-router/341)

### CODE REVIEW

File: src/router.js [hunk 1]
```diff
@@ -44,7 +53,7 @@ function createLocation({pathname = '', chain = [], params = {}, redirectFrom, r
       resolver
     )
   };
-}
+}e
```
Comment (vlukashov) [inline src/router.js]: This `e` looks like a typo. Was it intentional?

File: src/router.js [hunk 2]
```diff
@@ -439,7 +455,7 @@ export class Router extends Resolver {
           fireRouterEvent('location-changed', {router: this, location: this.location});
 
           if (shouldUpdateHistory) {
-            this.__updateBrowserHistory(context.pathname, context.redirectFrom);
+            this.__updateBrowserHistory(context, context.redirectFrom);
```
Comment (vlukashov) [inline src/router.js]: If the entire `context` is passed as a parameter, what's the main reason to pass the `context.redirectFrom` property as a separate second parameter?

File: src/router.js (hunk 1 above)
Comment (platosha) [reply src/router.js]: Fixed.

File: src/router.js (hunk 2 above)
Comment (platosha) [reply src/router.js]: Separation of concerns. This argument is named `replace`, it’s a boolean flag on whether the new location replaces or is pushed after the old one. The method is not supposed to decide this, I think, but the caller should.

**Pull request merged**